### ISSUE TITLE
Update vite and typescript in react-ts template

### ISSUE THREAD

**Issue by orellazri** [wailsapp/wails#3705]
### Is your feature request related to a problem? Please describe.

Vite is 2 major versions behind and TypeScript is 1 major version behind.
It causes issues when trying to use packages that require those dependencies to be more up to date.

### Describe the solution you'd like

Update the template's dependencies and files to the latest Vite and TypeScript versions.

### Describe alternatives you've considered

Update manually after creating the project. But it seems that upgrading to Vite 4 or above breaks hot reloading.

**Comment by drognisep** [comment]
I don't know if the same base template is used to create the templates included with the CLI, but the `vanilla` template Vite version is also [very far behind](https://vite.dev/releases) what is currently supported for fixes and security patches/backports.

At the time of this writing, updating to the latest version (`^8.0.3`) requires Node 20+, and Node 24 is the next [active LTS version](https://nodejs.org/en/about/previous-releases#release-schedule).

<img width="896" height="524" alt="Image" src="https://github.com/user-attachments/assets/7ffbe72e-8cb9-4028-8bf9-e64aafcaea89" />

**Comment by ninox14** [comment]
i manually bumped versions to vite 8.0.8 and react to latest 
seems to work fine besides dev script breaking when you change something on front end AFTER changes on back end 
vite dies with `Error: EPIPE: broken pipe, write` 

more info on error here https://chatgpt.com/share/69d98212-0ee0-8388-9178-d6ff4006372c

but if you just run front end vite server separately everything works fine

**Comment by xiaoxustudio** [comment]
Yes, it seems that the dependencies in the template are using very outdated versions. I can't help but wonder if this part is still being maintained?

**Comment by leaanthony** [comment]
These things just take time. There is nothing stopping you from updating them and submitting a PR 😄

**Comment by xiaoxustudio** [comment]
> These things just take time. There is nothing stopping you from updating them and submitting a PR 😄

I don't mean anything else. I'm just simply wondering if the template part will be updated (the other parts are still great). If you don't mind, I'd be more than willing to submit a pull request.
